src/pymor/algorithms/timestepping.py

```diff
@@ -230,6 +259,73 @@ def explicit_euler(A, F, U0, t0, t1, nt, mu=None, num_values=None):
     return R
 
 
+def implicit_midpoint_rule(A, F, M, U0, t0, t1, nt, mu=None, num_values=None, solver_options='operator'):
+    assert isinstance(A, Operator)
+    assert isinstance(F, (type(None), Operator, VectorArray))
+    assert isinstance(M, (type(None), Operator))
+    assert A.source == A.range
+    num_values = num_values or nt + 1
+    dt = (t1 - t0) / nt
+    DT = (t1 - t0) / (num_values - 1)
+
+    if F is None:
+        F_time_dep = False
+    elif isinstance(F, Operator):
+        assert F.source.dim == 1
+        assert F.range == A.range
+        F_time_dep = _depends_on_time(F, mu)
+        if not F_time_dep:
+            dt_F = F.as_vector(mu) * dt
+    else:
+        assert len(F) == 1
+        assert F in A.range
+        F_time_dep = False
+        dt_F = F * dt
+
+    if M is None:
+        from pymor.operators.constructions import IdentityOperator
+        M = IdentityOperator(A.source)
+
+    assert A.source == M.source == M.range
+    assert not M.parametric
+    assert U0 in A.source
+    assert len(U0) == 1
+
+    R = A.source.empty(reserve=nt+1)
+    R.append(U0)
+
+    if solver_options == 'operator':
+        options = A.solver_options
+    elif solver_options == 'mass':
+        options = M.solver_options
+    else:
+        options = solver_options
+
+    M_dt_A_impl = (M - A * (dt/2)).with_(solver_options=options)
```

Comment:
Maybe I'm confused. But shouldn't it be `M + A * (dt/2)` here when `A` is on the lhs (as for `implicit_euler`)?